2-Browsers/Week1/homework/ex5-catWalk/index.js

```diff
@@ -15,8 +15,35 @@
    for 5 seconds, and then replace the img with the original image and have it 
    continue the walk.
 -----------------------------------------------------------------------------*/
+
+const catImage = document.querySelector('img');
+catImage.style.left = '0px';
+const catImageWidth = catImage.width;
+const endOfScreen = window.innerWidth - catImageWidth;
+const midOfScreen = endOfScreen / 2;
+let catPosition = 0;
+let dancing = false;
+
 function catWalk() {
-  // TODO complete this function
-}
+  const startWalking = setInterval(function () {
+    catPosition += 10;
+    catImage.style.left = catPosition + 'px';
 
+    if (catPosition > endOfScreen) {
+      catPosition = 0;
+      dancing = false;
+    } else if (!dancing && catPosition >= midOfScreen) {
+      clearInterval(startWalking);
+      catImage.src =
+        'https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif?itemid=10561424';
+      setTimeout(() => {
+        dancing = true;
+        catImage.src =
+          'http://www.anniemation.com/clip_art/images/cat-walk.gif';
+        catWalk();
+      }, 5000);
+    }
+  }, 50);
+}
 // TODO execute `catWalk` when the browser has completed loading the page
+window.addEventListener('DOMContentLoaded', catWalk());
```

Comment:
Aside from the original image width, this functions perfectly. Very readable code also!